pkg/tiles/tile.go

```diff
@@ -0,0 +1,67 @@
+package tiles
+
+import (
+	"strconv"
+
+	buildings "github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles/buildings"
+)
+
+type Tile struct {
+	Cities
+	Roads
+	Fields
+	HasShield bool
+	Building  buildings.Bulding
+
+	//dać Building po prostu by skomponować
+
+	//not sure how to include undefied/null?
+	//meeple    Meeple
+}
+
+func (tile *Tile) Rotate(rotations uint) Tile {
```

Comment:
take self parameter as copy so it wouldn't be necessary to change this method while extending Tile struct